Source/AwsCommonRuntimeKit/http/HTTP2Settings.swift

```diff
@@ -0,0 +1,72 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import AwsCHttp
+
+/// Predefined settings identifiers (RFC-7540 6.5.2)
+/// Nil means use default values
```

Comment:
strange that Nil will be interpreted as a default value which I think could be something other than `nil`

Reply:
This is a common theme that if you don’t provide a value, the C APIs use the default. We decided not to replicate default values at multiple places because there is a high chance, we won’t update it everywhere if that value is updated in the future. In this case, the default values are mentioned in the RFC.